src/webgpu/web_platform/canvas/getCurrentTexture.spec.ts

```diff
@@ -4,24 +4,55 @@ Tests for GPUCanvasContext.getCurrentTexture.
 
 import { SkipTestCase } from '../../../common/framework/fixture.js';
 import { makeTestGroup } from '../../../common/framework/test_group.js';
+import { timeout } from '../../../common/util/timeout.js';
 import { assert, unreachable } from '../../../common/util/util.js';
 import { GPUTest } from '../../gpu_test.js';
 import { kAllCanvasTypes, createCanvas, CanvasType } from '../../util/create_elements.js';
 
+const kFormat = 'bgra8unorm';
+
 class GPUContextTest extends GPUTest {
   initCanvasContext(canvasType: CanvasType = 'onscreen'): GPUCanvasContext {
     const canvas = createCanvas(this, canvasType, 2, 2);
+    if (canvasType === 'onscreen') {
+      // To make sure onscreen canvas are visible
+      const onscreencanvas = canvas as HTMLCanvasElement;
+      onscreencanvas.style.position = 'fixed';
+      onscreencanvas.style.top = '0';
+      onscreencanvas.style.left = '0';
+      // Set it to transparent so that if multiple canvas are created, they are still visible.
+      onscreencanvas.style.opacity = '50%';
+      document.body.appendChild(onscreencanvas);
+    }
     const ctx = canvas.getContext('webgpu');
     assert(ctx instanceof GPUCanvasContext, 'Failed to get WebGPU context from canvas');
 
     ctx.configure({
       device: this.device,
-      format: 'rgba8unorm',
+      format: kFormat,
       usage: GPUTextureUsage.RENDER_ATTACHMENT | GPUTextureUsage.COPY_SRC,
     });
 
     return ctx;
   }
+
+  // Request a new "frame" based on the WebGPU canvas type.
+  requestNewFrameOnCanvasType(canvasType: CanvasType, ctx: GPUCanvasContext, fn: () => void) {
```

Comment:
I don't think this abstracts well over the two places we're using it right now [2]. One wants to do a frame loop so rAF makes sense, the other wants to run code at the beginning of the frame which is rPAF. I think I'd move the code back to those 2 places so we can modify them accordingly.